lib/integrations/awsoidc/create_ec2ice_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/google/go-cmp/cmp"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+type mockCreateEC2ICEClient struct {
+	name string
+	err  error
+}
+
+func (m mockCreateEC2ICEClient) CreateInstanceConnectEndpoint(ctx context.Context, params *ec2.CreateInstanceConnectEndpointInput, optFns ...func(*ec2.Options)) (*ec2.CreateInstanceConnectEndpointOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+
+	return &ec2.CreateInstanceConnectEndpointOutput{
+		InstanceConnectEndpoint: &ec2Types.Ec2InstanceConnectEndpoint{
+			InstanceConnectEndpointId: &m.name,
+		},
+	}, nil
+}
+
+func TestCreateEC2ICE(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("success", func(t *testing.T) {
+		mockCreateClient := &mockCreateEC2ICEClient{
+			name: "eice-123",
+		}
+		resp, err := CreateEC2ICE(ctx, mockCreateClient, CreateEC2ICERequest{
+			Cluster:         "c1",
+			IntegrationName: "i1",
+			SubnetID:        "subnet-id123",
+		})
+		require.NoError(t, err)
+		require.Equal(t, "eice-123", resp.Name)
+	})
+	t.Run("vpc reached the max allowed number of EC2 ICE", func(t *testing.T) {
```

Comment:
These two subtests don't share any common logic. I would either rewrite this as a table test or just break this up into two separate tests.